pkg/segment/structs/segstructs.go

```diff
@@ -578,9 +578,25 @@ func (ss *SegStats) Merge(other *SegStats) {
 
 	if ss.NumStats == nil {
 		ss.NumStats = other.NumStats
-		return
+	} else {
+		ss.NumStats.Merge(other.NumStats)
+	}
+	if ss.StringStats == nil {
+		ss.StringStats = other.StringStats
+	} else {
+		ss.StringStats.Merge(other.StringStats)
+	}
+}
+
+func (ss *StringStats) Merge(other *StringStats) {
+	if ss.StrSet != nil {
+		for key, value := range other.StrSet {
+			ss.StrSet[key] = value
+		}
+	}
+	if ss.StrList != nil {
+		ss.StrList = append(ss.StrList, other.StrList...)
```

Comment:
If `ss.StrSet == nil` or `ss.StrList == nil`, you should probably make the set/list and then insert the `other` values into it

Reply:
Made the changes, It will pick the other.StrList and other.StrSet when they are not nil.